backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/GwtGL30.java

```diff
@@ -0,0 +1,917 @@
+/*******************************************************************************
+ * Copyright 2022 See AUTHORS file.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.backends.gwt;
+
+import com.badlogic.gdx.graphics.GL20;
+import com.badlogic.gdx.graphics.GL30;
+import com.badlogic.gdx.graphics.Pixmap;
+import com.badlogic.gdx.utils.GdxRuntimeException;
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.core.client.JsArrayBoolean;
+import com.google.gwt.core.client.JsArrayInteger;
+import com.google.gwt.typedarrays.client.Uint8ArrayNative;
+import com.google.gwt.typedarrays.shared.ArrayBufferView;
+import com.google.gwt.typedarrays.shared.TypedArrays;
+import com.google.gwt.typedarrays.shared.Uint32Array;
+import com.google.gwt.webgl.client.WebGL2RenderingContext;
+import com.google.gwt.webgl.client.WebGLFramebuffer;
+import com.google.gwt.webgl.client.WebGLQuery;
+import com.google.gwt.webgl.client.WebGLSampler;
+import com.google.gwt.webgl.client.WebGLTexture;
+import com.google.gwt.webgl.client.WebGLTransformFeedback;
+import com.google.gwt.webgl.client.WebGLUniformLocation;
+import com.google.gwt.webgl.client.WebGLVertexArrayObject;
+
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.nio.FloatBuffer;
+import java.nio.HasArrayBufferView;
+import java.nio.IntBuffer;
+import java.nio.LongBuffer;
+
+/** @author Simon Gerst
+ * @author JamesTKhan */
+public class GwtGL30 extends GwtGL20 implements GL30 {
+	final IntMap<WebGLQuery> queries = IntMap.create();
```

Comment:
Can we make all/some of these attributes private/protected?

Reply:
Can do. I mostly followed with what GWTGL20 is doing here, which is not private but I can make them private here.

Reply:
Done.